### ISSUE TITLE
How to make `max-attributes-per-line` work only when over maximum line length

### ISSUE THREAD

**Issue by lyyourc** [vuejs/eslint-plugin-vue#378]
**Tell us about your environment**

* **ESLint Version:** `4.16.0`
* **eslint-plugin-vue Version:** `4.2.0`
* **Node Version:** `9.2.0`

**Please show your full configuration:**

<details>
  <summary> .eslintrc.js </summary>

  ```js
  module.exports = {
    extends: [
      'standard',
      'plugin:vue/recommended',
      'prettier',
    ],
    plugins: ['prettier'],
    parserOptions: {
      parser: 'babel-eslint',
      sourceType: 'module',
    },
    env: {
      browser: true,
      jest: true,
    },
    rules: {
      'prettier/prettier': 'error',

      // allow debugger during development
      'no-debugger': process.env.NODE_ENV === 'production' ? ['error'] : ['warn'],

      'no-param-reassign': ['error', { props: false }],

      'vue/html-closing-bracket-newline': [
        'error',
        {
          singleline: 'never',
          multiline: 'always',
        },
      ],

      'vue/html-closing-bracket-spacing': [
        'error',
        {
          startTag: 'never',
          endTag: 'never',
          selfClosingTag: 'always',
        },
      ],
    },
  }
  ```
</details>

**What did you do? Please include the actual source code causing the issue.**

```vue
<MyComponent lorem="1" ipsum="2" />

<MyComponent lorem="1" ipsum="2" @click="longlonglonglonglonglongFunctionName" @change="longlonglonglonglonglongFunctionName2" />
```

**What did you expect to happen?**
What i want is `max-attributes-per-line` rule should be available only when over **maximum line length** option, like `printWidth` defined in prettier.

For example, I set `printWidth: 80` in prettier: 
```vue
<MyComponent lorem="1"  ipsum="2" />

<MyComponent 
  lorem="1"
  ipsum="2"
  @click="longlonglonglonglonglongFunctionName" 
  @change="longlonglonglonglonglongFunctionName2"
/>
```

**What actually happened? Please include the actual, raw output from ESLint.**

As the default value of `multilple` option  in `max-attributes-per-line` rule is `1`, so the output will be:

```vue
<MyComponent
  lorem="1"
  ipsum="2"
/>

<MyComponent 
  lorem="1"
  ipsum="2"
  @click="longlonglonglonglonglongFunctionName" 
  @change="longlonglonglonglonglongFunctionName2"
/>
```

**Comment by mysticatea** [comment]
Thank you for the issue.

You can not configure `vue/max-attributes-per-line` rule with maximum line width.
The rule doesn't have any option for maximum line width currently.

ESLint rules are independent each other, so it's difficult.

**Comment by lyyourc** [comment]
Thanks replying so quickly.

We all hope`vue/max-attributes-per-line` can provide a `printWidth` option to support this feature,  
just like `--jsx-bracket-same-line` option in prettier.

**Comment by mysticatea** [comment]
Since the value of attributes is statements or expressions, so it's difficult to work together with other whitespace rules in ESLint.

Closing as no plan to add the option for now.

**Comment by octref** [comment]
I think this issue is better solved by a formatter. Vetur now supports prettyhtml and this is achievable:

![image](https://user-images.githubusercontent.com/4033249/49204459-ce65fb00-f360-11e8-88d8-28f5833a0e53.png)

**Comment by posva** [comment]
Will it make sense to add a new rule for this?

**Comment by harol-rodriguez** [comment]
use prettier

**Comment by codeflorist** [comment]
/rant

Sigh, i got rid of prettier, since eslint now is almost a valid replacement - wanting to get rid of all the headaches of using them simultaneously.

Now i discover that one of the biggest formatting-concerns in a vue project is not possible with eslint again.

I've tried the antfu package, now i'm on nuxt. Investing much time in switching.

I feel, i'm investing more time to get the whole linting/formatting shebang to work adequately than actual developing.

It's an endless stream of shortcomings and headaches...
